kafka/src/main/java/org/axonframework/extensions/kafka/eventhandling/producer/KafkaPublisher.java

```diff
@@ -296,14 +302,30 @@ public void shutDown() {
         }
 
         /**
-         * Set the Kafka {@code topic} to publish {@link EventMessage}s on. Defaults to {@code Axon.Events}.
+         * Set the Kafka {@code topic} to publish {@link EventMessage}s on. Defaults to {@code Axon.Events}. Should not
+         * be used together with setting the topicResolver.
```

Comment:
So...would we want to speak out a preference here? We could actually deprecate this `topic` method in favor of the `topicResolver` builder method.
This means the PR would be expanded somewhat, as the auto-configuration would also require us to use the `TopicResolver`.

But, what do you think? We could also do this in a follow-up PR, of course.

Reply:
That seems reasonable, also to maintain the code in the long run. And deprecation is much clearer than Javadoc + adds visibility to the new feature.